commonui/src/main/java/com/hyperwallet/android/ui/common/insight/HyperwalletInsight.java

```diff
@@ -18,34 +18,202 @@
 
 /**
  * Used for gathering the data necessary for the Insights analytics.
  */
 public class HyperwalletInsight {
     private static HyperwalletInsight sHyperwalletInsight;
+    private Executor mExecutor = Executors.newSingleThreadExecutor();
 
     private HyperwalletInsight() {
     }
 
+    /**
+     * Returns instance of  HyperwalletInsight or initializes it if it is null.
+     *
+     * @return singleton instance of HyperwalletInsight
+     */
     public static synchronized HyperwalletInsight getInstance() {
         if (sHyperwalletInsight == null) {
             sHyperwalletInsight = new HyperwalletInsight();
         }
         return sHyperwalletInsight;
     }
 
-    public void trackClick(@NonNull final Context context, @NonNull final String pageName,
-            @NonNull final String pageGroup, String link, Map<String, String> params) {
-
+    /**
+     * Initializes the Insight library using the given parameters.
+     *
+     * @param context      the context using Insight
+     * @param environment  the application's environment, ie. QA, UAT, PROD
+     * @param programToken the user's program token
+     * @param sdkVersion   the application's Hyperwallet SDK version
+     * @param apiUrl       the URL of the Insight server to send the data to
+     * @param userToken    the user's payee token
+     */
+    public void initializeInsight(final Context context, final String environment, final String programToken,
+            final String sdkVersion, final String apiUrl, final String userToken) {
+        Insight.initialize(context, apiUrl, userToken, environment, programToken, sdkVersion);
     }
 
+    /**
+     * Used to track an impression by a user. Typical example is when a user lands on a specific page.
+     *
+     * @param context   Context where the tracking is happening
+     * @param pageName  an arbituary pageName the user is currently viewing as defined by the app
+     * @param pageGroup an arbituary pageGroup the user is currently viewing as defined by the app
+     * @param params    additional information that can be added to make the tracking event more useful
+     */
     public void trackImpression(@NonNull final Context context, @NonNull final String pageName,
             @NonNull final String pageGroup, @NonNull final Map<String, String> params) {
 
+        if (Insight.getInsightTracker().isInitialized()) {
+            Insight.getInsightTracker().trackImpression(context, pageName, pageGroup, params);
+
+        } else {
+            mExecutor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    Hyperwallet.getDefault().getConfiguration(new HyperwalletListener<Configuration>() {
+                        @Override
+                        public void onSuccess(@Nullable Configuration result) {
+                            if (result != null) {
+                                onGetConfigurationSuccess(context, result);
```

Comment:
we can directly call `HyperwalletInsight.getInstance().initializeInsight(context, configuration)`